templates/documents/general_view.html

```diff
@@ -109,16 +109,24 @@
 <table valign="top" width="100%">
     <tr>
         <td>
+            <div style="margin-bottom: 6px;padding-bottom: 6px;border-bottom:3px solid gray;">
+            <a class="css_button">{$file->get_url_web()}</a>
+            <label for='chkShow' class="text" style="margin-left:20px;"><b>{xl t='View contents'}</b>
```

Comment:
To clarify, in terms of workflos:
1. As you are doing, show the document only by default.
2. Then if a user wants to see all the details, they can click and turn on the View Details toggle to show the details.